src/modules/launcher/Microsoft.Launcher/dllmain.cpp

```diff
@@ -364,15 +364,29 @@ void Microsoft_Launcher::parse_hotkey(PowerToysSettings::PowerToyValues& setting
 {
     try
     {
-        auto jsonHotkeyObject = settings.get_raw_json().GetNamedObject(JSON_KEY_PROPERTIES).GetNamedObject(JSON_KEY_OPEN_POWERLAUNCHER);
-        m_hotkey.win = jsonHotkeyObject.GetNamedBoolean(JSON_KEY_WIN);
-        m_hotkey.alt = jsonHotkeyObject.GetNamedBoolean(JSON_KEY_ALT);
-        m_hotkey.shift = jsonHotkeyObject.GetNamedBoolean(JSON_KEY_SHIFT);
-        m_hotkey.ctrl = jsonHotkeyObject.GetNamedBoolean(JSON_KEY_CTRL);
-        m_hotkey.key = static_cast<unsigned char>(jsonHotkeyObject.GetNamedNumber(JSON_KEY_CODE));
+        auto settingsObject = settings.get_raw_json();
+
+        if (settingsObject.HasKey(JSON_KEY_PROPERTIES))
+        {
+            auto jsonHotkeyObject = settingsObject.GetNamedObject(JSON_KEY_PROPERTIES).GetNamedObject(JSON_KEY_OPEN_POWERLAUNCHER);
+            m_hotkey.win = jsonHotkeyObject.GetNamedBoolean(JSON_KEY_WIN);
+            m_hotkey.alt = jsonHotkeyObject.GetNamedBoolean(JSON_KEY_ALT);
+            m_hotkey.shift = jsonHotkeyObject.GetNamedBoolean(JSON_KEY_SHIFT);
+            m_hotkey.ctrl = jsonHotkeyObject.GetNamedBoolean(JSON_KEY_CTRL);
+            m_hotkey.key = static_cast<unsigned char>(jsonHotkeyObject.GetNamedNumber(JSON_KEY_CODE));
+        }else
+        {
+            Logger::info("PT Launcher start shortcut is not present in settings. Use default shortcut instead.");
+            m_hotkey.win = false;
+            m_hotkey.alt = true;
+            m_hotkey.shift = false;
+            m_hotkey.ctrl = false;
+            m_hotkey.key = 32;
+        }
     }
     catch (...)
     {
+        Logger::error("Failed to initialize PT Launcher start shortcut");
```

Comment:
```suggestion
        Logger::error("Failed to initialize PT Run start shortcut");
```

Reply:
Why not also set it to Alt+Space here? This could happen if the json file doesn't have all the needed subkeys.

Reply:
@ivan100sic 
can you elaborate, please?

Reply:
If the .json file has the key `properties` but doesn't have `open_powerlauncher` subkey, or that subkey doesn't have `win`, `alt`, ... as subkeys, an exception will be thrown (and caught). It makes sense to set the shortcut to Alt+Space in that case.

Reply:
Right now, we assume that a valid settings file is either empty({}) or contains `properties.open_powerlauncher` with all modifier keys. It is better to get an error in logs if that structure changes because it will signify potential issues. We do not want to swallow an error.

Reply:
Of course, we should still log the error.

Reply:
Sorry, I misunderstood your suggestion.